src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManager.java

```diff
@@ -782,8 +812,15 @@ public RegisteredUserDTO updateUserObject(final RegisteredUser user) throws Inva
         if (!userToSave.getEmail().equals(existingUser.getEmail())) {
             try {
                 RegisteredUserDTO userToSaveDTO = mapper.map(userToSave, RegisteredUserDTO.class);
-                this.emailManager.sendEmailVerification(userToSaveDTO, userToSave.getEmailVerificationToken());
-            } catch (ContentManagerException | NoUserException e) {
+
+                Map<String, Object> emailTokens = ImmutableMap.of("verificationURL",
+                        this.generateEmailVerificationURL(userToSaveDTO, updatedUser.getEmailVerificationToken()));
```

Comment:
Somehow here we have gone from `userToSave.getEmailVerificationToken()` to `updatedUser.getEmailVerificationToken()`. This causes unhandled null pointer exceptions, because `updatedUser` never gets a verification token added to it. The `existingUser` does, as does `userToSave`; so I'm going to fix this to `userToSave`, which replicates the old behaviour and avoids the issue.